packages/pigeon/README.md

```diff
@@ -27,6 +27,10 @@ Custom classes, nested datatypes, and enums are also supported.
 
 Nullable enums in Objective-C generated code will be wrapped in a class to allow for nullability.
 
+By default, custom classes in Swift are defined as structs. 
+If you need classes instead - for recursive data, or Objective-C interop -
+use the @SwiftClass annotation when defining the class.
```

Comment:
Can we add a comment explaining why we need this? (e.g. struct doesn't support recursive definition)

Reply:
it is noted there already, is it not clear enough?

Reply:
> If you need classes instead - for recursive data, 
Can you add a note that Swift `struct` doesn't support recursively defined data type. 

> or Objective-C interop -
> use the @SwiftClass annotation when defining the class.

using `SwiftClass` isn't enough, we also need it to be `NSObject` subclass.